File: 2025-09-23 - Filas/Exemplo_C#/FilaInterface.cs
// ------------------------------------------------------------------------
// FilaInterface.cs
// Interface que define as operações básicas da fila.
// ------------------------------------------------------------------------
public interface FilaInterface
{
    bool Enqueue(Node novo);
    Node Dequeue();
    Node Front();
    bool IsEmpty();
    int Tamanho();
    void ImprimirLista();
    Node BuscarPorCodigo(string codigoDisciplina);
    bool Contains(string codigoDisciplina);
    bool Atualizar(string codigoDisciplina, string novoNome, int novaCarga);
    bool RemoverPorCodigo(string codigoDisciplina);
    void Clear();
}

File: 2025-09-23 - Filas/Exemplo_Java/FilaInterface.java
// ------------------------------------------------------------------------
// FilaInterface.java
// Interface que define as operações básicas da fila.
// ------------------------------------------------------------------------
public interface FilaInterface {
   boolean enqueue(Node novo);
   Node dequeue();
   Node front();
   boolean isEmpty();
   int tamanho();
   void imprimirLista();
   Node buscarPorCodigo(String codigoDisciplina);
   boolean contains(String codigoDisciplina);
   boolean atualizar(String codigoDisciplina, String novoNome, int novaCarga);
   boolean removerPorCodigo(String codigoDisciplina);
   void clear();
}
